lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -1060,6 +1063,23 @@ impl<'a> OptionalCheckScope<'a> {
     }
 }
 
+fn generate_get_token_account_space(mint: &Expr) -> proc_macro2::TokenStream {
+    quote! {
+        let mint_info = #mint.to_account_info();
```

Comment:
This line will not generate for me

`error: expected expression, found `let` statement`

When using init in a token account - 

```rs
    #[account(init,
        payer = signer,
        token::mint = mint,
        token::authority = signer,
    )]
    pub user_ata: Box<InterfaceAccount<'info, TokenAccount>>,
```

Changing to this for the if statement worked -

```rs
if #mint.to_account_info().owner == &anchor_spl::token_2022::ID
```

Reply:
Also fixed, thanks for the careful look